tasks/xpack/security/elasticsearch-security.yml

```diff
@@ -11,6 +11,23 @@
     - es_enable_xpack and '"security" in es_xpack_features'
     - (es_users is defined and es_users.file is defined) or (es_roles is defined and es_roles.file is defined) or (es_role_mapping is defined)
 
+#-----------------------------Create Bootstrap User-----------------------------------
+- name: Check if bootstrap password is set
```

Comment:
This section causes issues on my setup, when xpack is set to disabled due to the keystore not existing. Should this block of code ever be hit when xpack is not installed? Should we block it with a `when: (es_enable_xpack and '"security" in es_xpack_features')`

Reply:
@gregdurham just his this issue too.  You suggestion will work and think its the way to go.

Reply:
Yes I am making those changes, I encountered the same in further testing

Reply:
Thank you!

Reply:
Thanks!

Reply:
I am new  to Ansible and Elastic, and I ran the playbook on CentOS 7, i encountered the problem in elasticsearch-security.yml "Ensure x-pack conf directory exists (file)".  Do I have to create the keystore?   My error:

TASK [elasticsearch : Check if bootstrap password is set] ****************************************************************************************************
fatal: [esnode1]: FAILED! => {"changed": true, "cmd": ["/usr/share/elasticsearch/bin/elasticsearch-keystore", "list"], "delta": "0:00:01.361441", "end": "2018-01-23 09:13:37.358360", "failed": true, "msg": "non-zero return code", "rc": 65, "start": "2018-01-23 09:13:35.996919", "stderr": "", "stderr_lines": [], "stdout": "ERROR: Elasticsearch keystore not found. Use 'create' command to create one.", "stdout_lines": ["ERROR: Elasticsearch keystore not found. Use 'create' command to create one."]}